kmip/services/kmip_client.py

```diff
@@ -217,6 +217,7 @@ def open(self):
             suppress_ragged_eofs=self.suppress_ragged_eofs)
         self.protocol = KMIPProtocol(self.socket)
 
+        self.socket.settimeout(self.timeout)
```

Comment:
While I'm not opposed to the idea of better granularity when it comes to managing operation timeouts, I don't think that will work for this case. For the `socket` library, connection timeouts and socket timeouts are the same thing. Only one timeout value can be set, via `socket.settimeout(...)`, and it must be set before a connection is established to another socket. I do not believe that it can be reset once a connection is established; that would create unstable behavior. This timeout value applies to all subsequent socket operations, including `socket.connect`, `socket.sendall`, `socket.recv`, etc. (i.e., all `socket` operations used by the client).

Request messages in the KMIP 1.1 specification do not specify timeout limits for operations either. However, KMIP requests can specify the maximum response size (in bytes) that should be returned by the server (for large operation responses). It is also possible to execute an operation on a KMIP appliance as an asynchronous operation. In this case, the appliance should return an asynchronous correlation value (i.e., an operation token) that can be used to poll the appliance for the status of the operation. PyKMIP does not currently support either of these features.